Docs Restructuring: Usage Guidelines of the Starlite Application (#1153)
starlite-api/starlite: docs/lib/usage/starlite-app/app-hooks.rst

ASGI Events & Application Hooks
===============================

> In the previous introductory section of the documentations we briefly mentioned about the root

Reviewer: ```suggestion
In the previous introductory section of the documentations we briefly mentioned the root
```

> :class:`Starlite <starlite.app.Starlite>` class. This class accepts a bunch of `optional` arguments

Reviewer: ```suggestion
:class:`Starlite <starlite.app.Starlite>` class. This class accepts a severan `optional` arguments
```

Bunch is too informal

> & a couple of those arguments can be configured to control the behaviour of your Starlite application

Reviewer: please don't use `&` at all. This is a reapting issue. 

I think we also agreed on using American spelling. Hence -> `behaviour` -> `behavior` etc.

Author: Yeah, my bad about using `&`, I keep using it out of habit :sweat_smile:. I will do a search & replace before marking this PR as "ready for review".

> based on certain ASGI events. This section of the documentations take a deeper dive into how you

Reviewer: ```suggestion
. This section of the documentations take a deeper dive into how you
```

> based on certain ASGI events. This section of the documentations take a more deeper dive into how you

Reviewer: ```suggestion
based on certain ASGI events. This section of the documentations takes a deeper dive into how you
```

> can use those hooks to configure your application throughout its lifespan.

Reviewer: ```suggestion
can use these hooks to configure your application throughout its lifespan.
```

> Brief Overview of Application Hooks and Their Respective Events

Reviewer: ```suggestion
Application Hooks Overview
```

-------------------------------------------------------------

> All ASGI webservers (like `Uvicorn <https://www.uvicorn.org>`_ or

Reviewer: ```suggestion
ASGI webservers (like `Uvicorn <https://www.uvicorn.org>`_ or
```

> `Daphne <https://github.com/django/daphne>`_ and such) emits certain events through the lifespan of the

Reviewer: ```suggestion
`Daphne <https://github.com/django/daphne>`_ and such) emit certain events through the lifespan of the
```

> application they are invoked with. And Starlite can hook into those events to perform certain tasks as

Reviewer: this is not english `application they are invoked with`. Rephrase this please.

`And Starlite can hook into those events to perform certain tasks as`

those -> these.
And -> should not be the beginning of a sentence
`And Starlite` -> hanging sentence.

Rewrite this please

intended for that particular event.

> These hooks accepts a `list of callables` - either sync/async functions, methods or class instances and

Reviewer: Ok, look there are grammatical errors on basically every line. This PR cannot proceed like this. 

```suggestion
These hooks accept a `list of callables` - either sync/async functions, methods or class instances and
```

Following sentence switches from these to those etc. 

Use something like grammarly, or other tools to fix the grammer and please have someone with native level english proof and rewrite this.
